ioctl/cmd/ws/wsprojectupdate.go

```diff
@@ -44,32 +50,62 @@ var (
 
 func init() {
 	wsProjectUpdate.Flags().Uint64P("project-id", "i", 0, config.TranslateInLang(_flagProjectIDUsages, config.UILanguage))
-	wsProjectUpdate.Flags().StringP("project-uri", "u", "", config.TranslateInLang(_flagProjectURIUsages, config.UILanguage))
-	wsProjectUpdate.Flags().StringP("project-hash", "v", "", config.TranslateInLang(_flagProjectHashUsages, config.UILanguage))
+	wsProjectUpdate.Flags().StringP("project-config-file", "u", "", config.TranslateInLang(_flagProjectConfigFileUsages, config.UILanguage))
+	wsProjectUpdate.Flags().StringP("project-config-hash", "v", "", config.TranslateInLang(_flagProjectConfigHashUsages, config.UILanguage))
 
 	_ = wsProjectCreate.MarkFlagRequired("project-id")
-	_ = wsProjectCreate.MarkFlagRequired("project-uri")
-	_ = wsProjectCreate.MarkFlagRequired("project-hash")
+	_ = wsProjectCreate.MarkFlagRequired("project-config-file")
 }
 
-func updateProject(projectID uint64, uri, hash string) error {
+func updateProject(projectID uint64, filename, hashstr string) (string, error) {
+	uri, hashv, err := upload(wsProjectIPFSEndpoint, filename, hashstr)
```

Comment:
`createProject` used `upload`, here also use `upload`? wouldn't it overwrite existing project config?

Reply:
the `url` and `hash` will be written to contract and the content will store in IPFS. so the existing project would not overwrite.